### PR TITLE
feat: rename input to text input and add props

### PR DESCRIPTION
**Pull request by gabriel-piltzer-trimble**
### :page_facing_up: Summary of Changes

This PR renames `modus-wc-input` to `modus-wc-text-input` to make room for `modus-wc-number-input` (future).

It adds props that are in the current Modus Web Components that seems like they could be useful.

### :thought_balloon: Type of Change

- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)
- [ ] Configuration change (modifies scaling or properties of existing infrastructure & services)
- [ ] Documentation change
- [x] Other

### :clipboard: Test Plan

100% code covered

### :white_check_mark: Self Code Review Checklist

PR authors and reviewers, please verify that all of these items have been completed.

- [x] My code is clean and readable
- [x] I followed standard conventions
- [x] Component code is WCAG 2.2 compliant (if applicable)
- [ ] I have made theme files changes (if applicable)
- [x] I have made documentation changes (if applicable)
- [x] I have made Storybook changes including usage documentation (if applicable)
- [x] I have tested the component thoroughly in Storybook including RTL rendering (if applicable)

### :link: Azure DevOps Work Item

[AB#566232](https://dev.azure.com/ViewpointVSO/74194628-0b6e-4b03-95a4-2d06c069dcbe/_workitems/edit/566232)

### CODE REVIEW

File: src/components/atoms/modus-wc-text-input/modus-wc-text-input.tsx [hunk 1]
```diff
@@ -26,6 +26,27 @@ export class ModusWcInput {
    */
   @Prop() ariaLabel!: string;
 
+  /**
+   * Controls automatic capitalization in inputted text.
+   */
+  @Prop() autoCapitalize?:
```
Comment (stuart-rimel) [inline src/components/atoms/modus-wc-text-input/modus-wc-text-input.tsx:32]: This prop doesn't appear to be working when testing through storybook

**Review by stuart-rimel — COMMENTED** [review]
Looks good! Just one comment

File: src/components/atoms/modus-wc-text-input/modus-wc-text-input.tsx (hunk 1 above)
Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-text-input/modus-wc-text-input.tsx:32]: https://caniuse.com/?search=autocapitalize

image

Comment (gabriel-piltzer-trimble) [reply src/components/atoms/modus-wc-text-input/modus-wc-text-input.tsx:32]: Likely only affects virtual keyboards

Comment (stuart-rimel) [reply src/components/atoms/modus-wc-text-input/modus-wc-text-input.tsx:32]: Per [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/autocapitalize) this doesn't affect physical keyboard typing which invalidates my comment.

**Review by stuart-rimel — APPROVED** [review]

**Pull request merged**